src/scalar.rs

```diff
@@ -1336,27 +1305,19 @@ mod test {
         0, 0, 0, 0, 15, 0, 0, 0, 0, 15, 0, 0, 0, 0, 15, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0,
     ];
 
-    static LARGEST_ED25519_S: Scalar = Scalar {
-        bytes: [
-            0xf8, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-            0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-            0xff, 0xff, 0xff, 0x7f,
-        ],
-    };
-
-    static CANONICAL_LARGEST_ED25519_S_PLUS_ONE: Scalar = Scalar {
+    const BASEPOINT_ORDER_MINUS_ONE: Scalar = Scalar {
         bytes: [
-            0x7e, 0x34, 0x47, 0x75, 0x47, 0x4a, 0x7f, 0x97, 0x23, 0xb6, 0x3a, 0x8b, 0xe9, 0x2a,
-            0xe7, 0x6d, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
-            0xff, 0xff, 0xff, 0x0f,
+            0xec, 0xd3, 0xf5, 0x5c, 0x1a, 0x63, 0x12, 0x58, 0xd6, 0x9c, 0xf7, 0xa2, 0xde, 0xf9,
+            0xde, 0x14, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x10,
         ],
     };
 
-    static CANONICAL_LARGEST_ED25519_S_MINUS_ONE: Scalar = Scalar {
+    static LARGEST_ED25519_S: Scalar = Scalar {
```

Comment:
If `clamp_integer` were `const fn`, couldn't this be computed by clamping `LARGEST_UNREDUCED_SCALAR`?

Reply:
Yup good catch